### ISSUE TITLE
Autowrap tests skipped on Travis

### ISSUE THREAD

**Issue by oscarbenjamin** [sympy/sympy#17959]
The tests in `sympy/external/tests/test_autowrap.py` are skipped in the optional dependency build on Travis. This is because here
https://github.com/sympy/sympy/blob/2662c098cb19fb48c821ca890d6d664a2efda555/sympy/external/tests/test_autowrap.py#L14-L21
something fails and we get `f2pyworks=False`.

I have tested not catching the exception from autowrap to see what it looks like in #17950.

It seems that there is something not quite right with the way that gfortran is set up in Travis but I'm not sure exactly what that is.

**Comment by asmeurer** [comment]
There's a good chance this is a bug in autowrap itself. If the Fortran compiler on Travis works correctly (and it ought to, since we are just installing it from apt), then autowrap should be able to use it properly. I suspect the issue has to do with how it gets the gfortran shared libraries, although I haven't dug into it to confirm that.

**Comment by oscarbenjamin** [comment]
Is it possible that there could be a mixup between things installed by apt and conda?

(I've never used autowrap and also don't have much experience with conda)

**Comment by asmeurer** [comment]
Yes, my guess is it's trying to use the shared library directory from conda instead of the system one. It should ideally detect the right one to use.
